### PR TITLE
Add support for sdax variants when generating BOM

### PR DESCRIPTION
**Pull request by jackHay22**
Adds support for SDAX variants during BOM generation. 
Uses test file: https://hub.allspice.io/NoIndexTests/parallela-sdax/src/branch/main/variant-test-1.sdax

### Example BOM without variant:
|Description|Designator|Name|Part Number|Quantity|
|---|---|---|---|---|
|1K|R1|Resistor_Test||1|
|4.7K|R2|Resistor_Test-4.7k||1|
|1uF|U1|Capacitor_Test||1|
|||||1|
|||||1|
|1K|R5|Resistor_Test||1|
|1K|R6|Resistor_Test||1|
|1K|R3|Resistor_Test||1|
|1K|R4|Resistor_Test||1|
|1K|R11|Resistor_Test||1|
|1K|R12|Resistor_Test||1|
|1K|R13|Resistor_Test||1|
|1K|R14|Resistor_Test||1|

### Example BOM with variant TESTVAR:
|Description|Designator|Name|Part Number|Quantity|
|---|---|---|---|---|
|1K|R1|Resistor_Test||1|
|4.7K|R2|Resistor_Test-4.7k||1|
|4.7K|U1|Resistor_Test-4.7k||1|
|||||1|
|||||1|
|4.7K|R5|Resistor_Test-4.7k||1|
|1K|R6|Resistor_Test||1|
|1K|R4|Resistor_Test||1|
|1K|R11|Resistor_Test||1|
|1K|R12|Resistor_Test||1|
|1K|R13|Resistor_Test||1|
|1K|R14|Resistor_Test||1|

### CODE REVIEW

**Comment by kdumontnu** [comment]
@jackHay22 are there any DNI components in parallela test?

**Comment by jackHay22** [comment]
> @jackHay22 are there any DNI components in parallela test?

Yes, `R3`.

File: allspice/utils/list_components.py [hunk 1]
```diff
@@ -393,7 +391,9 @@ def list_components_for_orcad(
         to each component to store the name of the component.
     """
 
-    components = _list_components_multi_page_schematic(allspice_client, repository, dsn_path, ref)
+    components = _list_components_multi_page_schematic(
```
Comment (shrik450) [inline allspice/utils/list_components.py:394]: Given this logic is entirely shared, it might make sense to take the variant arg and pass it on anyway, with the documentation noting that if it shows up in Hub we can handle it here. That way we're not held back by a py-allspice update once we support OrCAD variants in crysknife. Your call!

File: allspice/utils/list_components.py [hunk 2]
```diff
@@ -461,24 +467,53 @@ def _list_components_multi_page_schematic(
...
+
+            else:
+                component_attributes = _component_attributes_multi_page(component)
+                components.append(component_attributes)
 
     return _filter_blank_components(components, allspice_client.logger)
 
 
-def _fetch_generated_json(repo: Repository, file_path: str, ref: Ref) -> dict:
+def _fetch_generated_json(
```
Comment (shrik450) [inline allspice/utils/list_components.py:510]: Can you rename this to `retry_not_yet_generated` (or something like that)? It could also live in a file of its own in the `utils` module, because it ends up being something I re-implement whenever I use py-allspice with the generated JSON.

File: allspice/apiobject.py [hunk 3]
```diff
@@ -1224,6 +1225,29 @@ def get_generated_svg(self, content: Union[Content, str], ref: Optional[Ref] = N
         data = Util.data_params_for_ref(ref)
         return self.allspice_client.requests_get_raw(url, data)
 
+    def get_generated_projectdata(
+        self, content: Union[Content, str], ref: Optional[Ref] = None
+    ) -> dict:
```
[the remark this answers is not in the document]
Comment (jackHay22) [reply allspice/apiobject.py:1230]: Is it worth adding an additional test file for this? The existing PCB will return empty project data but in order to test retries we'd need to add an sdax file.

**Review by kdumontnu — APPROVED** [review]
👍
